Drivers/Src/CANbus.c

```diff
@@ -39,18 +39,29 @@ static void CANbus_CountIncoming(void) {
 /**
  * @brief   Initializes the CAN system
  * @param   loopback	: if we should use loopback mode (for testing)	
+ * @param 	faultState  : determines whether to implement Rx and Tx interrupts
  * @return  None
  */
-void CANbus_Init(bool loopback) {
+void CANbus_Init(bool loopback, bool faultState) {
+	if(!faultState){
+		RTOS_BPS_MutexCreate(&CANbus_TxMutex, "CAN TX Lock");
+		RTOS_BPS_MutexCreate(&CANbus_RxMutex, "CAN RX Lock");
+		RTOS_BPS_SemCreate(&CANbus_MailSem4, "CAN Mailbox Semaphore", 3); // # of mailboxes
+		RTOS_BPS_SemCreate(&CANbus_ReceiveSem4, "CAN Queue Counter Semaphore", 0);
+	}
 
-	RTOS_BPS_MutexCreate(&CANbus_TxMutex, "CAN TX Lock");
-	RTOS_BPS_MutexCreate(&CANbus_RxMutex, "CAN RX Lock");
-	RTOS_BPS_SemCreate(&CANbus_MailSem4, "CAN Mailbox Semaphore", 3); // # of mailboxes
-	RTOS_BPS_SemCreate(&CANbus_ReceiveSem4, "CAN Queue Counter Semaphore", 0);
 	// Initialize and pass interrupt hooks
     BSP_CAN_Init(CANbus_CountIncoming, CANbus_Release, loopback);
```

Comment:
I'm confused, why do you pass in a faultstate variable into the driver level function for CAN initialization, and then use the tasks fault flag in the BPS level? Wouldn't it make more sense to do it the other way around (or the same way for both) so the layers don't cross? The BSP level should not be dependent on the tasks level IMO.

Reply:
fixed